assets/js/src/integrations/classic.js

```diff
@@ -40,15 +40,26 @@ export function classicTracking(
 	} );
 
 	// Handle runtime cart events.
+	const oldAddedToCart = document.body.onadded_to_cart;
 	/**
 	 * Track the custom add to cart event dispatched by WooCommerce Core
 	 *
 	 * @param {Event}         e         - The event object
 	 * @param {Object}        fragments - An object containing fragments of the updated cart.
 	 * @param {string}        cartHash  - A string representing the hash of the cart after the update.
 	 * @param {HTMLElement[]} button    - An array of HTML elements representing the add to cart button.
+	 * @param {Array}         rest      - An array of additional arguments passed to the event handler.
 	 */
-	document.body.onadded_to_cart = ( e, fragments, cartHash, button ) => {
+	document.body.onadded_to_cart = (
+		e,
+		fragments,
+		cartHash,
+		button,
+		...rest
+	) => {
+		if ( typeof oldAddedToCart === 'function' ) {
+			oldAddedToCart( e, fragments, cartHash, button, ...rest );
```

Comment:
If the goal is to minimize the impact on other extensions, it's also possible to carry `this` over correctly. For example, I would expect the following function to be called with [the `body` node](https://github.com/woocommerce/woocommerce/blob/8.9.3/plugins/woocommerce/client/legacy/js/frontend/add-to-cart.js#L116) as `this` instead of the `window` object, even though `this` isn't normally used.

```php
add_action(
	'wp_head',
	function () {
		wp_add_inline_script(
			'woocommerce-google-analytics-integration',
			'document.body.onadded_to_cart = function() { console.log( this, ...arguments ); };',
			'before',
		);
	},
);
```